### PR TITLE
Add Testerina Function Mocking and Object Mocking details

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Update release notes

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md [hunk 1]
```diff
@@ -703,6 +703,29 @@ ballerina bindgen [(-cp|--classpath) <classpath>...]
...
+
+```
+@test:MockFn {
+    functionName : "<function_to_mock>"
+}
+test:MockFunction mockObj = new();
+```
+
+Function mocking is brought about by using the following functions
+- `test:when(mockObj)` is used to initialize the mocking capability within a particular test case
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md:718]:
```suggestion
- The `test:when(mockObj)` is used to initialize the mocking capability within a particular test case
```

File: downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md [hunk 2]
```diff
@@ -703,6 +703,29 @@ ballerina bindgen [(-cp|--classpath) <classpath>...]
...
+- Allows us to use the associated mocking functions like `call()`, `thenReturn()`, `withArguments()`
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md:719]:
```suggestion
- This allows you to use the associated mocking functions like `call()`, `thenReturn()`, and `withArguments()`.
```

File: downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md [hunk 3]
```diff
@@ -703,6 +703,29 @@ ballerina bindgen [(-cp|--classpath) <classpath>...]
...
+
+#### Object Mocking
+
+Object mocking enables controlling the values of member variables and the behavior of the member functions of an object. 
+
+- Introduced creating a test double, which provides an equivalent mock in place of the real object
+
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md:726]:
```suggestion

```

Remove this new line between the bullet points.

File: downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md [hunk 4]
```diff
@@ -703,6 +703,29 @@ ballerina bindgen [(-cp|--classpath) <classpath>...]
...
+- Introduced capability of stubbing the member function or member variable
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md:727]:
```suggestion
- Introduced the capability of stubbing the member function or member variable
```

File: downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md [hunk 5]
```diff
@@ -703,6 +703,29 @@ ballerina bindgen [(-cp|--classpath) <classpath>...]
...
+
+Function mocking is brought about by using the following functions
+- `test:when(mockObj)` is used to initialize the mocking capability within a particular test case
+- Allows us to use the associated mocking functions like `call()`, `thenReturn()`, `withArguments()`
+
+#### Object Mocking
+
+Object mocking enables controlling the values of member variables and the behavior of the member functions of an object. 
+
+- Introduced creating a test double, which provides an equivalent mock in place of the real object
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md:725]:
```suggestion
- Introduced the ability to create a test double, which provides an equivalent mock in place of the real object
```

What is test double? Is it "double test"?

File: downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md [hunk 6]
```diff
@@ -703,6 +703,29 @@ ballerina bindgen [(-cp|--classpath) <classpath>...]
 
 Introducing the Mocking API for object and function mocking.
 
+#### Function Mocking
+
+`MockFunction` object added to handle function mocking. The `@test:MockFn` annotation attached to the MockFunction specifies the function to mock. MockFunction objects are defined as such.
```
[the remark this answers is not in the document]
Comment (Ibaqu) [reply downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md]: Updated

**Pull request merged**